src/internals.jl

```diff
@@ -66,8 +66,7 @@ is_chunkable(::AbstractChunks) = true
 _set_minsize(minsize::Nothing, _) = 1
 function _set_minsize(minsize::Integer, collection_length::Integer)
     minsize < 1 && throw(ArgumentError("minsize must be >= 1"))
-    minsize > collection_length && throw(ArgumentError("minsize must be <= length(collection)"))
-    return minsize
+    return max(min(minsize, collection_length), 1)
```

Comment:
No need for the `max()` here, since an error is throws already for `minsize < 1`?

Reply:
ah, ok, if `collection_length == 0` we get one empty chunk...